backend/copr_backend/actions.py

```diff
@@ -112,32 +113,27 @@ class Createrepo(Action):
     def run(self):
         self.log.info("Action createrepo")
         data = json.loads(self.data["data"])
-        ownername = data["ownername"]
-        projectname = data["projectname"]
         project_dirnames = data["project_dirnames"]
         chroots = data["chroots"]
-        appstream = data["appstream"]
-        devel = data["devel"]
-
         result = BackendResultEnum("success")
 
-        done_count = 0
         for project_dirname in project_dirnames:
             for chroot in chroots:
-                self.log.info("Creating repo for: %s/%s/%s",
-                              ownername, project_dirname, chroot)
-                repo = os.path.join(self.destdir, ownername,
-                                    project_dirname, chroot)
-                try:
-                    os.makedirs(repo)
-                    self.log.info("Empty repo so far, directory created")
-                except FileExistsError:
-                    pass
-
-                if not call_copr_repo(repo, appstream=appstream, devel=devel,
-                                      logger=self.log):
+                # Fake a Job interface
+                # So far it has been useful to pass `Job` objects to all storage
+                # methods. Unfortunatelly jobs represent only builds and not
+                # actions for which the interface is different. Until we better
+                # know what we want, I am faking the Job interface.
+                job = Munch(data)
+                job.project_owner = data["ownername"]
+                job.project_name = project_dirname
+                job.chroot = chroot
+                job.uses_devel_repo = data["devel"]
+
+                storage = storage_for_job(job, self.opts, self.log)
+                success = storage.init_project(job)
```

Comment:
No big deal either, but it would be much more convenient if had a per-project
storage object:

```
self.storage = get_storage(owner, project_name)
for dirname in project_dirnames:
    self.storage.init_dirname(dirname, chroots)
```

We wouldn't have to re-instantiate the storage object fore very dirname/chroot.